src/client/Microsoft.Identity.Client/Internal/Requests/ClientCredentialRequest.cs

```diff
@@ -42,6 +43,14 @@ protected override async Task<AuthenticationResult> ExecuteAsync(CancellationTok
 
             AuthenticationResult authResult = null;
 
+            if (AuthenticationRequestParameters.Authority is AadAuthority aadAuthority)
+            {
+                if (aadAuthority.IsCommonOrganizationsOrConsumersTenant())
```

Comment:
We ban this flow on /common and /organizations because these tenants really mean "I don't know the tenant, let AAD figure it out for me", which only works for user flows. 

I do not think we should bad /consumers, as this is an actual tenant id (98...)

Reply:
indeed, we shouldn't and sorry I wasn't clear. I meant the error message isn't consistent with the test method.

Reply:
oh ok. will update